src/shell.js

```diff
@@ -232,10 +232,14 @@ if (ENVIRONMENT_IS_NODE) {
 #if ENVIRONMENT_MAY_BE_SHELL || ASSERTIONS
 if (ENVIRONMENT_IS_SHELL) {
 
-#if ENVIRONMENT
-#if ASSERTIONS
-  if ((typeof process === 'object' && typeof require === 'function') || typeof window === 'object' || typeof importScripts === 'function') throw new Error('not compiled for this environment (did you build to HTML and try to run it not on the web, or set ENVIRONMENT to something - like node - and run it someplace else - like on the web?)');
-#endif
+#if ENVIRONMENT && ASSERTIONS
+  if ((typeof process === 'object'
+       && typeof process.versions === 'object'
+       && typeof process.versions.node === 'string'), /* NODE */
```

Comment:
the comma here looks wrong, or I don't know how to read this line...?
```suggestion
       && typeof process.versions.node === 'string') /* NODE */
```